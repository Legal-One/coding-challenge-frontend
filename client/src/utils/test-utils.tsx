import React, { ReactElement, ReactNode } from 'react'
import { render as rtlRender } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import callCenterReducer from '../features/callcenter/callcenter-slice'

function render(
  ui: JSX.Element,
  {
    initialState = {},
    store = configureStore({
      reducer: {
        callCenter: callCenterReducer
      },
      preloadedState: initialState
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }: { children: JSX.Element }) {
    const history = createMemoryHistory()
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    )
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

export * from '@testing-library/react'
export { render }
