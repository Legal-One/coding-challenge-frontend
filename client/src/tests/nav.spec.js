import { mount } from "@vue/test-utils";
import NavComponent from "../components/NavComponent.vue";
it('renders correctly', () => {
    const wrapper = mount(NavComponent)
    expect(wrapper.element).toMatchSnapshot()
})

it('has a link to dashboard', () => {
    const wrapper = mount(NavComponent)
     expect(wrapper.get('router-link').isVisible()).toBe(true)
})