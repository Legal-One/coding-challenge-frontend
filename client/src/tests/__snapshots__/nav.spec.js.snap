// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  class="flex items-start justify-start flex-wrap bg-teal-500 p-6"
>
  <div
    class="block lg:hidden"
  >
    <button
      class="flex items-center px-3 py-2 border rounded text-teal-200 border-teal-400 hover:text-white hover:border-white"
    >
      <svg
        class="fill-current h-3 w-3"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Menu
        </title>
        <path
          d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
        />
      </svg>
    </button>
  </div>
  <div
    class="w-full block justify-start lg:flex items-start lg:w-auto"
  >
    <div
      class="text-sm lg:flex-grow"
    >
      <router-link
        class="block mt-4 lg:inline-block lg:mt-0 text-teal-200 hover:text-white mr-4 text-lg"
        to="/"
      >
         General Report 
      </router-link>
      <!-- &lt;router-link to="/" class="block mt-4 lg:inline-block lg:mt-0 text-teal-200 hover:text-white mr-4 text-lg"&gt;
        Agent List
      &lt;/router-link&gt; -->
    </div>
  </div>
</nav>
`;
